### DISCUSSION TITLE
Performing node level Regression/classification without Repetition

### DISCUSSION THREAD

**Discussion by SourabhRanade** [pyg-team/pytorch_geometric#6694] [Q&A] [answered]
I have created a custom graph based dataset using the tutorials mentioned at the following url of Pygeomtric: [Creating Your Own Datasets](https://pytorch-geometric.readthedocs.io/en/latest/tutorial/create_dataset.html). 
The dataset consists of multiple graphs with different number of nodes in them. 
Now,I want to perform regression/classification task on each of these graphs and predict the nodes order in every graph. This essentially means I want every node to have a uique,sequential and non-reapating output. On a crude level if there are 3 nodes in one graph then  I would like every node's output to be classified as 0,1,2 or 1,2,3.  
The dataset that I have created has such labels for every node in every graph.

As I am very new to PyG , I wanted to know if I can refer to any material or can make something like this possible ?
Thank you in advance.

**Comment by rusty1s** [comment, marked as answer]
Mh, this task is new to me, but it looks like you want to perform ranking over nodes. There exists a lot of ranking losses, e.g., BPR loss which you can use. Afterwards, you can validate the order of ranking via `pred.argmax()`.

**Reply by SourabhRanade** [reply]
@rusty1s thanks a lot for the answer. Yes, 'ranking' would be the right word to use I suppose. Are these ranking losses already supported in Pytorch Geometric or would I need to implement them on my own ?
If they are documented/implemented could you please guide me to those resources? I tried finding them on my own but could not find anything specifically from Pytorch or PyG.

**Reply by rusty1s** [reply]
I think BPR is one of the most popular used in [LightGCN](https://arxiv.org/pdf/2002.02126.pdf). We also implemented it as part of our `LightGCN` module.

**Reply by SourabhRanade** [reply]
@rusty1s , thank you so much ... I am having a look at the implementation and the paper to better my understanding. 
However, @rusty1s I still haven't understood, how I can construct a model that can handle graphs with different number of nodes in each graph where the num_classes are essentially dependent on the nodes in a graph.
